### PR TITLE
Deprecate Python module 'partitioning'. Move functionality to 'community'.

### PR DESCRIPTION
**Pull request by fabratu**
This PR also adds a second version of `community.getImbalance` for both C++ and Python. Not sure, whether the former definition in the community module should be kept.

### CODE REVIEW

File: networkit/cpp/community/GraphClusteringTools.cpp [hunk 1]
```diff
@@ -14,8 +14,17 @@ float getImbalance(const Partition &zeta) {
...
+}
+
+float getImbalance(const Partition &zeta, const Graph &graph) {
+    float avg = std::ceil(
+        (float)graph.numberOfNodes()
+        / (float)zeta.numberOfSubsets());
+    std::vector<count> clusterSizes = zeta.subsetSizes();
+    float maxClusterSize = (float) *std::max_element(clusterSizes.begin(),
+            clusterSizes.end());
+    return static_cast<float>(maxClusterSize / avg);
```
Comment (angriman) [inline networkit/cpp/community/GraphClusteringTools.cpp:27]: Both variables are float, the cast should not be necessary
```suggestion
    return maxClusterSize / avg;
```

File: networkit/cpp/community/GraphClusteringTools.cpp [hunk 2]
```diff
@@ -14,8 +14,17 @@ float getImbalance(const Partition &zeta) {
     std::vector<count> clusterSizes = zeta.subsetSizes();
     float maxClusterSize = (float) *std::max_element(clusterSizes.begin(),
             clusterSizes.end());
-    float imbalance = maxClusterSize / avg;
-    return imbalance;
+    return static_cast<float>(maxClusterSize / avg);
```
Comment (angriman) [inline networkit/cpp/community/GraphClusteringTools.cpp:17]:
```suggestion
    return maxClusterSize / avg;
```

File: networkit/cpp/community/GraphClusteringTools.cpp (hunk 1 above)
Comment (angriman) [reply networkit/cpp/community/GraphClusteringTools.cpp:27]: The two functions are very similar, the only difference is the computation of `avg`. I suggest having a single private function (e.g., `computeImbalance`) and using it in the two public `getImbalace`.

**Review by angriman — APPROVED** [review]

**Comment by fabratu** [comment]
Merging this early since it should be part of the upcoming release.

**Pull request merged**